### ISSUE TITLE
Sigrok Windows build

### ISSUE THREAD

**Issue by dzarda** [dojoe/Twonkie#2]
Hey man and thanks for the initiative.

I'm considering building this, but what's the current usage with Sigrok? Do we have a way of piping the bitstream into Sigrok for example? `sigrok-cli -i -`

**Comment by dojoe** [comment]
Hey, the Twinkie authors wrote a sigrok driver for Twinkie which also works for Twonkie. Details are in the original Twinkie docs here: https://www.chromium.org/chromium-os/twinkie#TOC-Using-as-a-PD-Packet-Sniffer

The Twinkie patches haven't been upstreamed so far, so if the .debs on there don't work for you you'll have to build from source. I have a relatively recent Windows build too if you need that - just need to find a place to host it.

**Comment by dzarda** [comment]
Yeah having a Windows build available would be much appreciated. Wanna upload it to the GH release?

**Comment by dojoe** [comment]
Yep - I was reluctant at first since it's not technically built from the repo but what the hay =) There you go!

**Comment by dzarda** [comment]
What path do you see of getting the functionality upstream?

**Comment by dojoe** [comment]
Someone's gotta send the patches to the sigrok project and petition the maintainers to pull them in. I was planning to take care of that but won't have the time in the near future - if you have some spare time I'd be delighted if you could give it a shot. Shouldn't take much convincing now that there are three separate open source projects supporting the protocol :)
I adapted Vincent's patches to the latest sigrok during my Windows build work already so they should be ready to go (modulo change requests by the maintainers); I can share them later today.

**Comment by dojoe** [comment]
Ugh, I take that back :/ It's not as easy unfortunately - I had to add some ugly hacks to make sigrok work with a recent libusb on Windows, that'll take some cleanups and proper implementation before I can release it. I don't have the time to do that right now, but I'll get to it eventually. In the meantime, you have binaries :)

**Comment by madcrow** [comment]
Hey Joachim, congrats on this awesome project!
I am considering building some v2 devices. I plan to use them under Windows 10.
Do I understand correctly that I can use the official upstream Sigrok / PulseView windows builds (as they already include the PD protocol decoder), and I just need to add the Twonkie driver? Or do the Sigrok / PulseView main executables / libs / whatever need to be patched as well? Can you share your current "Twonkie Windows client package"? It's OK, if you are not ready to share the sources yet, the binaries should be more than enough for now.
Does the current release of Sigrok / PulseView support decoding EPR-related messages as well?

**Comment by dojoe** [comment]
Hey @madcrow - thanks for your interest!
Unfortunately sigrok requires a recompile of at least libsigrok to support new devices, and the Windows build is kinda finicky. I lost the build environment I had to a Windows reinstall but short term you can use the builds I posted in https://github.com/dojoe/Twonkie/releases/tag/twonkie-20210310.
If you'd like to contribute it would be ace if you could port vpalatin's patches in https://github.com/vpalatin/libsigrok/tree/twinkie-0.5.0-win-WIP to the latest sigrok source and get them upstreamed. This may require some re-implementing since the current patches use a feature of libusb that is not supported on Windows. I don't remember how I worked around that. I'd love to get that working myself but I'm pretty short on time :/
